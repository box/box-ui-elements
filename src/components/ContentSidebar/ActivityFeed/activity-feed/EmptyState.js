/**
 * @flow
 * @file Component for Activity feed empty state
 */

import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import LoadingIndicator from 'box-react-ui/lib/components/loading-indicator/LoadingIndicator';
import IconActivityFeedEmptyState from '../icons';

import messages from '../../../messages';

type Props = {
    isLoading?: boolean,
    showCommentMessage?: boolean,
};

const EmptyState = ({ isLoading, showCommentMessage }: Props): React.Node => (
    <div className="bcs-activity-feed-empty-state">
        <IconActivityFeedEmptyState />
        {isLoading ? (
            <LoadingIndicator />
        ) : (
            <div className="bcs-empty-state-cta">
                <FormattedMessage {...messages.noActivity} />
                {showCommentMessage ? (
                    <aside>
                        <FormattedMessage
                            {...messages.noActivityCommentPrompt}
                        />
                    </aside>
                ) : null}
            </div>
        )}
    </div>
);

export default EmptyState;
