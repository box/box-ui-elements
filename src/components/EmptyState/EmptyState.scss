.buik-empty {
    align-items: center;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: center;
}
