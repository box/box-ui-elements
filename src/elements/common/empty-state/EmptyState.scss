.be-empty {
    display: flex;
    flex: 1;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
}
