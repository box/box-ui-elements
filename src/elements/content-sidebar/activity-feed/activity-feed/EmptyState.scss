.bcs-EmptyState {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.bcs-EmptyState-cta {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.bcs-EmptyState-cta-detail {
    font-weight: bold;
}

.bcs-EmptyState-cta-message {
    width: 200px;
    margin-top: 6px;
    text-align: center;
}

.bcs-EmptyState-illustration {
    margin-top: 60px;
    margin-bottom: 15px;
}
