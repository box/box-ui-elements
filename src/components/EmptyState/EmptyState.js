/**
 * @flow
 * @file Empty state component
 * @author Box
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import ErrorEmptyState from 'box-react-ui/lib/icons/states/ErrorEmptyState';
import FolderEmptyState from 'box-react-ui/lib/icons/states/FolderEmptyState';
import SelectedItemsEmptyState from 'box-react-ui/lib/icons/states/SelectedItemsEmptyState';
import SearchEmptyState from 'box-react-ui/lib/icons/states/SearchEmptyState';
import messages from '../messages';
import {
    VIEW_ERROR,
    VIEW_FOLDER,
    VIEW_SEARCH,
    VIEW_SELECTED,
} from '../../constants';

import './EmptyState.scss';

type Props = {
    view: View,
    isLoading: boolean,
};

const EmptyState = ({ view, isLoading }: Props) => {
    let type;
    const message =
        isLoading && view === VIEW_FOLDER ? (
            <FormattedMessage {...messages.loadingState} />
        ) : (
            <FormattedMessage {...messages[`${view}State`]} />
        );

    switch (view) {
        case VIEW_ERROR:
            type = <ErrorEmptyState />;
            break;
        case VIEW_SELECTED:
            type = <SelectedItemsEmptyState />;
            break;
        case VIEW_SEARCH:
            type = <SearchEmptyState />;
            break;
        default:
            type = <FolderEmptyState />;
            break;
    }

    return (
        <div className="be-empty">
            {type}
            <div>{message}</div>
        </div>
    );
};

export default EmptyState;
