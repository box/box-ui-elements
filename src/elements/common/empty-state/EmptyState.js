/**
 * @flow
 * @file Empty state component
 * @author Box
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import ErrorEmptyState from '../../../icons/states/ErrorEmptyState';
import FolderEmptyState from '../../../icons/states/FolderEmptyState';
import SelectedItemsEmptyState from '../../../icons/states/SelectedItemsEmptyState';
import SearchEmptyState from '../../../icons/states/SearchEmptyState';
import messages from '../messages';
import { VIEW_ERROR, VIEW_FOLDER, VIEW_METADATA, VIEW_SEARCH, VIEW_SELECTED } from '../../../constants';
import type { View } from '../../../common/types/core';

import './EmptyState.scss';

type Props = {
    isLoading: boolean,
    view: View,
};

const EmptyState = ({ view, isLoading }: Props) => {
    let type;
    const message =
        isLoading && (view === VIEW_FOLDER || view === VIEW_METADATA) ? (
            <FormattedMessage {...messages.loadingState} />
        ) : (
            <FormattedMessage {...messages[`${view}State`]} />
        );

    switch (view) {
        case VIEW_ERROR:
            type = <ErrorEmptyState />;
            break;
        case VIEW_SELECTED:
            type = <SelectedItemsEmptyState />;
            break;
        case VIEW_SEARCH:
            type = <SearchEmptyState />;
            break;
        default:
            type = <FolderEmptyState />;
            break;
    }

    return (
        <div className="be-empty">
            {type}
            <div>{message}</div>
        </div>
    );
};

export default EmptyState;
