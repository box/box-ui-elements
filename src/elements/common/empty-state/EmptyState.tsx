import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { Files, FolderFloat, HatWand, OpenBook } from '@box/blueprint-web-assets/illustrations/Medium';

import messages from '../messages';
import { VIEW_ERROR, VIEW_FOLDER, VIEW_METADATA, VIEW_SEARCH, VIEW_SELECTED } from '../../../constants';
import type { View } from '../../../common/types/core';

import './EmptyState.scss';

export interface EmptyStateProps {
    isLoading: boolean;
    view: View;
}

const EmptyState = ({ view, isLoading }: EmptyStateProps) => {
    let type;
    const message =
        isLoading && (view === VIEW_FOLDER || view === VIEW_METADATA) ? (
            <FormattedMessage {...messages.loadingState} />
        ) : (
            <FormattedMessage {...messages[`${view}State`]} />
        );

    switch (view) {
        case VIEW_ERROR:
            type = <HatWand />;
            break;
        case VIEW_SELECTED:
            type = <Files />;
            break;
        case VIEW_SEARCH:
            type = <OpenBook />;
            break;
        default:
            type = <FolderFloat />;
            break;
    }

    return (
        <div className="be-empty">
            {type}
            <div>{message}</div>
        </div>
    );
};

export default EmptyState;
