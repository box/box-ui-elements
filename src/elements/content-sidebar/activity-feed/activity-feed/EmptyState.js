/**
 * @flow
 * @file Component for Activity feed empty state
 */

import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import ActivityFeedEmptyStateIllustration from '../illustrations/ActivityFeedEmptyStateIllustration';
import EmptyStatePreviewActivity140 from '../../../../illustration/EmptyStatePreviewActivity140';
import messages from '../../../common/messages';
import './EmptyState.scss';

type Props = {
    showAnnotationMessage?: boolean,
    showCommentMessage?: boolean,
};

const EmptyState = ({ showAnnotationMessage, showCommentMessage }: Props): React.Node => {
    const showActionMessage = showAnnotationMessage || showCommentMessage;
    const actionMessage = showAnnotationMessage
        ? messages.noActivityAnnotationPrompt
        : messages.noActivityCommentPrompt;

    return (
        <div className="bcs-EmptyState">
            <div className="bcs-EmptyState-illustration">
                {showAnnotationMessage ? <ActivityFeedEmptyStateIllustration /> : <EmptyStatePreviewActivity140 />}
            </div>
            <div className="bcs-EmptyState-cta">
                <FormattedMessage {...messages.noActivity}>
                    {(text: string) => <span className="bcs-EmptyState-cta-detail">{text}</span>}
                </FormattedMessage>
                {showActionMessage && (
                    <aside className="bcs-EmptyState-cta-message">
                        <FormattedMessage {...actionMessage} />
                    </aside>
                )}
            </div>
        </div>
    );
};

export default EmptyState;
