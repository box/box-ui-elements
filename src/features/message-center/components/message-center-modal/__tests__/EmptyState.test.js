// @flow
import * as React from 'react';
import { shallow } from 'enzyme';

import EmptyState from '../EmptyState';

describe('components/message-center/components/message-center-modal/EmptyState', () => {
    function getWrapper() {
        return shallow(<EmptyState />);
    }

    test('should render correctly', async () => {
        expect(getWrapper()).toMatchSnapshot();
    });
});
