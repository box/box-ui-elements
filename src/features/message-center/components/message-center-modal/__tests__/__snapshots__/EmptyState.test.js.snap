// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/message-center/components/message-center-modal/EmptyState should render correctly 1`] = `
<ErrorState>
  <FormattedMessage
    defaultMessage="There are no posts for this category at the moment."
    id="be.messageCenter.noPosts"
  />
</ErrorState>
`;
